src/FF/source/FFMatrixUtil.cc

```diff
@@ -208,6 +208,123 @@ void load_matrix_data(pdb::PDBClient &pdbClient, string path,
   pdbClient.flushData(errMsg);
 }
 
+template <class T>
+void loadMatrixGenericFromFile(pdb::PDBClient &pdbClient, std::string path,
+                      pdb::String dbName, pdb::String setName, int totalX, int totalY, int blockX,
+                      int blockY,
+                      string &errMsg, int size, bool partitionByCol) {
+  if (path.size() == 0) {
+    throw runtime_error("Invalid filepath: " + path);
+  }
+
+
+
+  // open the input file
+  ifstream inFile(path);
+
+  double val;
+  int total = 0;
+  pdb::makeObjectAllocatorBlock(size * 1024 * 1024, true);
+
+  pdb::Handle<pdb::Vector<pdb::Handle<T>>> storeMatrix =
```

Comment:
Professor, is it okay if the generic type "T" is a Handle here? 
Asking this considering our recent performance improvements by removing this for this task.

Reply:
Handle must be used with a PC Object, which is any object that inherits from the pdb::Object class.
So Handle<Handle> won't work. This is a bit different with the C-style pointers where you can have pointer to pointer.

Reply:
Got it